keras_cv/models/backbones/vit/vit_backbone.py

```diff
@@ -0,0 +1,176 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ViT (Vision Transformer) models for Keras.
+Reference:
+  - [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929v2)
+    (ICLR 2021)
+  - [How to train your ViT? Data, Augmentation, and Regularization in Vision Transformers](https://arxiv.org/abs/2106.10270)
+    (CoRR 2021)
+"""  # noqa: E501
+
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.layers import TransformerEncoder
+from keras_cv.layers.vit_layers import PatchingAndEmbedding
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.vit.vit_backbone_presets import backbone_presets
+from keras_cv.models.backbones.vit.vit_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ViTBackbone(Backbone):
+    """Instantiates the ViT architecture.
+
+    Reference:
+        - [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929v2)
+        (ICLR 2021)
+    This function returns a Keras {name} model.
+
+    The naming convention of ViT models follows: ViTSize_Patch-size
+        (i.e. ViTS16).
+    The following sizes were released in the original paper:
+        - S (Small)
+        - B (Base)
+        - L (Large)
+    But subsequent work from the same authors introduced:
+        - Ti (Tiny)
+        - H (Huge)
+
+    The parameter configurations for all of these sizes, at patch sizes 16 and
+    32 are made available, following the naming convention laid out above.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+
+        include_rescaling: bool, whether to rescale the inputs. If set to
+            True, inputs will be passed through a `Rescaling(1/255.0)` layer.
+        input_shape: optional shape tuple, defaults to (224, 224, 3).
+        input_tensor: optional Keras tensor (i.e. output of `layers.Input()`)
+            to use as image input for the model.
+        patch_size: the patch size to be supplied to the Patching layer to turn
+            input images into a flattened sequence of patches.
+        transformer_layer_num: the number of transformer encoder layers to stack
```

Comment:
I wonder if we couldn't use the KerasNLP layer OOB since the CV implementation seems quite standard.

Reply:
This might also be something to coordinate wiith @DavidLandup0 since he is merging in Transformer layers

Reply:
My 2 cents is that we should keep them separate. IMO, it's better to have a bit of duplicated code (i.e. one class) than to try to make cross-lib dependencies. Since we'll be adding a few CV-specific transformer encoders and attention types which will necessarily have to be part of KCV only, fragmenting the implementations doesn't feel like the best idea to me.

We _could_ try to see if a vanilla transformer encoder could become a part of core Keras though, and have both being used in KNLP and KCV. As far as I can tell, there's demand for the class as part of the main framework and it would make sense to move a layer to it if both KNLP and KCV would use it, since that means that it's not specific enough to sit in only one lib?

Reply:
There's also a slightly different impl (KNLP seems to use one layer norm, i.e. `if self.normalize_first` and `if not self.normalize_first`) while this layer uses two (one before all ops and one after the residual). Merging the implementations would wreck the weights of one lib